tests/test_optimizers.py

```diff
@@ -309,33 +311,84 @@ def test_transpose_dequantize_with_axis(self, shape, perm_input, perm_output):
                                    model_proto, remaining_transpose_num=0)
 
     @parameterized.expand([
-        ((2, 3, 4, 5), [1, 2, 1, 2], (1, 2, 2, 1), [0, 2, 3, 1], [0, 3, 1, 2]),
-        ((2, 3, 4, 5, 6), [1, 2, 1, 2, 1], (1, 1, 2, 1, 2), [0, 2, 3, 4, 1], [0, 4, 1, 2, 3]),
+        ([2, 3, 4, 5], [1, 2, 1, 2], [0, 2, 3, 1], [0, 3, 1, 2]),
+        ([2, 3, 4, 5, 6], [1, 2, 1, 2, 1], [0, 2, 3, 4, 1], [0, 4, 1, 2, 3]),
     ])
-    @check_opset_min_version(10, "Slice in opset 10 can accept dymaic 'start' and 'ends'")
-    def test_transpose_slice(self, input_shape, slice_size, output_shape, perm_input, perm_output):
-        starts = np.array([0] * len(input_shape), dtype=np.int64)
-        ends = np.array(slice_size, dtype=np.int64)
-        axes = np.array(list(range(len(input_shape))), dtype=np.int64)
-        node1 = helper.make_node("Transpose", ["X"], ["Y"], perm=perm_input, name="trans_1")
-        node2 = helper.make_node("Slice", ["Y", "starts", "ends", "axes"], ["Z"], name="relu")
-        node3 = helper.make_node("Transpose", ["Z"], ["Z1"], perm=perm_output, name="trans_2")
+    @check_opset_max_version(9, "Slice in opset 9 and takes 'axes, 'start' and 'ends' as attributes")
+    def test_transpose_slice(self, input_shape, slice_size, perm_input, perm_output):
+        axes_combinations = []
+        axes = list(range(len(input_shape)))
+        for i in range(1, len(input_shape) + 1):
+            axes_combinations.extend(list(itertools.combinations(axes, i)))
```

Comment:
Do we really need to test every single combination?  That's like 32 runs per test.  Glad to know they all pass, but I think now we can safely just add an arg for parameterized "axes" and fill in a few tests, get rid of the loop here.

Reply:
sure, I will add parametrized axes.